### PR TITLE
Create 0019-ui-tests-against-dev.md

### PR DESCRIPTION
**Pull request by dneed-nimble**

### CODE REVIEW

**Comment by nwarms** [comment]
I have ADR 18 on my branch, can you bump the number please?

File: docs/adrs/0019-ui-tests-against-dev.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Status
+
+Accepted
+
+## Context
+
+Previously, we had two types of Cypress tests for our UI:
+
+- **Deployment Tests**: These pointed to the live URL of our Dev environment.
```
Comment (dynamictulip) [inline docs/adrs/0019-ui-tests-against-dev.md:13]: Also to the Test environment. They were lightweight and the sole purpose was to ensure that a deployment hadn't broken the whole system

File: docs/adrs/0019-ui-tests-against-dev.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+1. **Closer to Accurate Data**: The Dev environment is connected to the **Dev Academies Database**, which mirrors production changes to it's structure (not neccesarily it's data) more closely than the isolated database we previously used. By utilising this, we ensure that our tests are running against data that is representative of what will be in other environments. This helps us catch issues that arise due to real-world changes in the data.
+
+2. **Simplification**: The process becomes significantly simpler. By removing the need to maintain and manage the fake database and isolated containers, we reduce the complexity of our testing pipeline.
+
+3. **Reduced Maintenance Overhead**: Removing the fake database means we no longer need to maintain its data model or ensure that it reflects changes occurring in real databases.
+
+## Consequences
+
+- **Reduced Control over Test Data**: By moving away from an isolated database, we lose some control over the ability to insert and manipulate data in specific ways. While this was useful in certain edge cases, it also came with a significant overhead.
```
Comment (dynamictulip) [inline docs/adrs/0019-ui-tests-against-dev.md:32]: The reduced control goes slightly further than this - without knowing what should be in any particular field, we can't test that any field contains the data we expect.

For example, the contacts page has a number of names and emails against different roles but we can no longer test that the right person is listed as say the Accounting officer. We also can't test that at a unit level and it is hard to test manually so our best line of defence is extra vigilance in PR reviews.

This means that our UI tests have to be less prescriptive with the side effect of having less coverage - "can I go to the academies ofsted page and does the table have some rows in it" rather than "does the academies ofsted page for this trust display 13 rows with this exact data in it and these before/after joining tags"

**Review by dynamictulip — APPROVED** [review]

**Review by nwarms — APPROVED** [review]
